test/integration/controllermanager/shoot/state/state_test.go

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package state_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/shoot/state/finalizer"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("ShootState controller test", func() {
+	var (
+		shoot          *gardencorev1beta1.Shoot
+		shootState     *gardencorev1beta1.ShootState
+		targetSeedName = "target-seed"
+	)
+
+	BeforeEach(func() {
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+				Namespace:    testNamespace.Name,
+				Labels:       map[string]string{testID: testRunID},
+			},
+			Spec: gardencorev1beta1.ShootSpec{
+				SecretBindingName: ptr.To("secretbinding"),
+				CloudProfileName:  ptr.To("cloudprofile1"),
+				Region:            "europe-central-1",
+				Provider: gardencorev1beta1.Provider{
+					Type: "foo-provider",
+					Workers: []gardencorev1beta1.Worker{
+						{
+							Name:    "cpu-worker",
+							Minimum: 3,
+							Maximum: 3,
+							Machine: gardencorev1beta1.Machine{
+								Type: "large",
+							},
+						},
+					},
+				},
+				DNS: &gardencorev1beta1.DNS{
+					Domain: ptr.To("some-domain.example.com"),
+				},
+				Kubernetes: gardencorev1beta1.Kubernetes{
+					Version: "1.25.1",
```

Comment:
```suggestion
					Version: "1.32.0",
```

Reply:
Thanks for the suggestion, updated it to `1.32.0` in my latest updates.